src/webdriver.rs

```diff
@@ -188,3 +188,65 @@ where
         }
     }
 }
+
+#[cfg(feature = "persist")]
+pub mod persist {
+    use std::marker::PhantomData;
+
+    use serde::{Deserialize, Serialize};
+
+    use crate::{
+        common::types::SessionId,
+        error::WebDriverResult,
+        http::connection_async::WebDriverHttpClientAsync,
+        session::{spawn_session_task, WebDriverSession},
+        webdriver::{GenericWebDriver, WebDriver},
+    };
+
+    /// The persisted session that can be serialized and deserialized.
+    ///
+    /// # TODO
+    ///
+    ///  - What is `WebDriverConfig` used for? Do we need to persist it as well?
+    ///
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    pub struct PersistedSession {
+        session_id: SessionId,
+        capabilities: serde_json::Value,
+    }
+
+    impl PersistedSession {
+        pub async fn connect<T>(
+            self,
+            remote_server_addr: &str,
+        ) -> WebDriverResult<GenericWebDriver<T>>
+        where
+            T: WebDriverHttpClientAsync + 'static,
+        {
+            let conn = T::create(remote_server_addr)?;
+            // TODO: Check if session exists?
+            let tx = spawn_session_task(Box::new(conn));
+
+            let driver = GenericWebDriver {
+                session: WebDriverSession::new(self.session_id, tx),
+                capabilities: self.capabilities,
+                quit_on_drop: true,
+                phantom: PhantomData,
+            };
+
+            Ok(driver)
+        }
+    }
+
+    impl WebDriver {
```

Comment:
Does this work? My thinking is that it should be `impl GenericWebDriver` instead.

Also, would you be happy to add a test or doctest for the persist() -> connect() round trip?

Reply:
Hmm, it probably should be `GenericWebDriver`, but this definitely compiles and even runs!

I can add a test, when I have some time. It might take a while, but I guess we can keep the PR open until then.

Reply:
I don't think a test would be difficult at all. Have a look at the doctests for WebDriverCommands etc. All this would require is to create a session, persist it to a file, destroy the original, then reload the session from the file and issue some command to it. I think that would be sufficient as a minimal way to sanity check this feature.

Happy to keep the PR open :)

Reply:
Note that you can use `docker-compose up -d` to run the demo web app that all doctests use. It will be accessible at `http://localhost:8000` when it runs. By looking at the other tests you will see how this is being used. It's a fairly simple yet robust way to test everything against an actual selenium instance and browser.

Once there are doctests for this I'd also like this to be added to the automated build (Github Actions) so that this feature is regularly tested with every build, but I can add that later :)